rusty-machine/examples/nnet-and_gate.rs

```diff
@@ -0,0 +1,89 @@
+extern crate rusty_machine;
+extern crate rand;
+
+use rand::{random, Open01};
+use std::vec::Vec;
+
+use rusty_machine::learning::nnet::{NeuralNet, BCECriterion};
+use rusty_machine::learning::toolkit::regularization::Regularization;
+use rusty_machine::learning::optim::grad_desc::StochasticGD;
+
+use rusty_machine::linalg::matrix::Matrix;
+use rusty_machine::learning::SupModel;
+
+// AND gate
+fn main() {
+    println!("AND gate learner sample:");
+
+    const THRESHOLD: f64 = 0.7;
+
+    const SAMPLES: usize = 10000;
+    println!("Generating {} training data and labels...", SAMPLES as u32);
+
+    let mut input_data = Vec::with_capacity(SAMPLES * 2);
+    let mut label_data = Vec::with_capacity(SAMPLES);
+
+    for _ in 0..SAMPLES {
+        let Open01(left) = random::<Open01<f64>>();
+        let Open01(right) = random::<Open01<f64>>();
+        input_data.push(left);
+        input_data.push(right);
+        if left > THRESHOLD && right > THRESHOLD {
+            label_data.push(1.0);
+        } else {
+            label_data.push(0.0)
+        }
+    }
+
+    let inputs = Matrix::new(SAMPLES, 2, input_data);
+    let targets = Matrix::new(SAMPLES, 1, label_data);
+
+    let layers = &[2, 1];
+    let criterion = BCECriterion::new(Regularization::L2(0.));
+    let mut model = NeuralNet::new(layers, criterion, StochasticGD::default());
+
+    println!("Training...");
+    model.train(&inputs, &targets);
+
+    let test_cases = vec![
+        0.0, 0.0,
+        0.0, 1.0,
+        1.0, 1.0,
+        1.0, 0.0,
+        ];
+    let expected = vec![
+        0.0,
+        0.0,
+        1.0,
+        0.0,
+        ];
+    let test_inputs = Matrix::new(test_cases.len() / 2, 2, test_cases);
+    let res = model.predict(&test_inputs);
+
+    println!("Evaluation...");
+    let mut hits = 0;
+    let mut misses = 0;
+    // Evaluation
+    println!("Got\tExpected");
+    //for (idx, prediction) in res.into_vec().iter().enumerate() {
```

Comment:
On this one, I fixed it, but the original intention was to show two ways to evaluate the network output (we have no softmax and co right now). The problem, as you saw, was that I committed exactly the same code!

As it was just a tiny "plus", I have just removed the commented section.